lib/worker.coffee

```diff
@@ -44,3 +51,25 @@ module.exports = (text, attributes, options) ->
       stack: stack
 
   callback()
+
+registerBlocksPositions = (doc) ->
+  # Use Ruby API:
+  # because `doc.findBy()`` doesn't yet accept a filter function as a parameter.
+  # https://github.com/asciidoctor/asciidoctor.js/issues/282
+  blocks = Opal.block_send doc, 'find_by', (b) -> b.$lineno() isnt Opal.nil
+
+  linesMapping = {}
+  for block, index in blocks
+    lineno = block.$lineno()
+
+    if typeof lineno is 'number'
```

Comment:
You no longer need to check if lineno is a number since we are pre-filtering the results.